### ISSUE TITLE
upgrade sql-alchemy version to at least 1.4.29

### ISSUE THREAD

**Issue by ihvol-freenome** [googleapis/python-bigquery-sqlalchemy#477]
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

 **Is your feature request related to a problem? Please describe.**
yes. current requirement for `sql-alchemy` is `>=1.2.0,<=1.4.27`  and `sql-alchemy` version `1.4.27` has [this issue](https://github.com/sqlalchemy/sqlalchemy/issues/7496). we can't upgrade sql-alchemy because of this dependency.
 **Describe the solution you'd like**
Please upgrade the `sql-alchemy` to at least `1.4.29` version which has a fix for the issue described above. 
 **Describe alternatives you've considered**
alternative workaround would be `# type: ignore` to ignore the entire file, but that doesn't comply with our code quality policy and we can't use it.
 **Additional context**
Here is the full trace...
```
version: 0.961
Traceback (most recent call last):
  File "mypy/semanal.py", line 5350, in accept
  File "mypy/nodes.py", line 1165, in accept
  File "mypy/semanal.py", line 2109, in visit_assignment_stmt
  File "mypy/semanal.py", line 2395, in apply_dynamic_class_hook
  File "/fn/lib/python3.9/site-packages/sqlalchemy/ext/mypy/plugin.py", line 145, in _dynamic_class_hook
    obj = ctx.api.named_type("__builtins__.object")
  File "mypy/semanal.py", line 4720, in named_type
  File "mypy/semanal.py", line 4687, in lookup_fully_qualified
AssertionError: __builtins__.object
```

**Comment by meredithslota** [comment]
This is temporarily blocked by some failing tests, please see here for additional info: https://github.com/googleapis/python-bigquery-sqlalchemy/issues/385 I'm going to close this issue as a duplicate of that issue, and I hope we can prioritize this soon.
